src/main/java/hudson/security/LDAPSecurityRealm.java

```diff
@@ -970,11 +975,32 @@ public String getName() {
         @Override
         public GroupDetailsImpl mapAttributes(String dn, Attributes attributes) throws NamingException {
             LdapName name = new LdapName(dn);
-            String groupName = fixGroupname(String.valueOf(name.getRdn(name.size() - 1).getValue()));
+            String groupName = fixGroupname(extractGroupName(name, attributes));
             return new GroupDetailsImpl(dn, groupName);
         }
-    }
 
+        static String extractGroupName(LdapName name, Attributes attributes) throws NamingException {
+            final String CN = "cn";
+            String groupName = StringUtils.EMPTY;
+            if (!GET_CN_FROM_ATTRIBUTES) {
+                groupName = String.valueOf(name.getRdn(name.size() - 1).getValue());
+            } else {
+                for (NamingEnumeration ae = attributes.getAll(); ae.hasMore(); ) {
+                    Attribute attr = (Attribute) ae.next();
+                    if (CN.equals(attr.getID())) {
+                        for (NamingEnumeration e = attr.getAll(); groupName.equals(StringUtils.EMPTY);) {
```

Comment:
Do you also need `e.hasMore()` in the condition?

Reply:
done